oauth2_provider/oauth2_backends.py

```diff
@@ -44,7 +45,26 @@ def _extract_params(self, request):
             del headers['wsgi.errors']
         if 'HTTP_AUTHORIZATION' in headers:
             headers['Authorization'] = headers['HTTP_AUTHORIZATION']
-        body = urlencode(request.POST.items())
+
+        return headers
+
+    def _extract_body(self, request):
+        """
+        Extracts the POST body from the Django request object
+        :param request: The current django.http.HttpRequest object
+        :return: provided POST parameters
+        """
+        return request.POST.items()
+
+    def _extract_params(self, request):
+        """
+        Extract parameters from the Django request object. Such parameters will then be passed to
+        OAuthLib to build its own Request object. The body should be encoded using OAuthLib urlencoded
+        """
+        uri = self._get_escaped_full_path(request)
+        http_method = request.method
+        headers = self._extract_headers(request)
+        body = urlencode(self._extract_body(request))
```

Comment:
The only thing I would change is remove the _ in the ``_extract_headers`` and ``_extract_body`` methods. The underscore prefix usually mean that the method is "private" and one shouldn't override it.

Maybe we need to define which is the public API of the class and set the name accordingly.